dso/model.py

```diff
@@ -133,26 +143,31 @@ class ComponentArtefactId:
     artefact_kind: ArtefactKind | None = None
     references: list[typing.Self] = dataclasses.field(default_factory=list)
 
-    def as_frozenset(self) -> frozenset[str]:
-        props = (
+    @property
+    def key(self) -> str:
```

Comment:
since key is quite imporant, and its intended use might not be obvious, I think a short docstr might be adequate (state, in particular, that key should identify this artefact-metadata uniquely, and is intended to be used as pk in underlying db)